src/bootupd.rs

```diff
@@ -124,8 +124,9 @@ pub(crate) fn install(source_root: &str, dest_root: &str) -> Result<()> {
 pub(crate) fn get_components() -> Vec<Box<dyn Component>> {
     let mut components: Vec<Box<dyn Component>> = Vec::new();
 
+    // Unwrap safety - this component is built in for these targets
     #[cfg(any(target_arch = "x86_64", target_arch = "aarch64"))]
-    components.push(Box::new(efi::EFI::new()));
+    components.push(component::new_from_name("EFI").expect("efi"));
```

Comment:
I did like the previous state of this. Why did you change the approach? Would the `EFI::new_component` suggestion below help?

Reply:
Hm...is your objection the new unwrap?  I reworked it to call directly again.

Reply:
Not directly, I think it was overall on the "lookup via hardcoded string label". It's somehow subjective and minor, thus sorry I couldn't articulate it better. Current form looks good :+1:.